src/python/pants/engine/legacy/graph.py

```diff
@@ -425,12 +422,39 @@ class TopologicallyOrderedTargets:
   hydrated_targets: HydratedTargets
 
 
+class InvalidOwnersOfArgs(Exception):
+  pass
+
+
 @dataclass(frozen=True)
 class OwnersRequest:
   """A request for the owners (and optionally, transitive dependees) of a set of file paths."""
-  sources: Tuple
+  sources: Tuple[str, ...]
   include_dependees: IncludeDependeesOption
 
+  def validate(self, *, pants_bin_name: str) -> None:
```

Comment:
Should this be a `__post_init__`?

Reply:
Originally, it was. But the issue with `__post__init__` is that it can't take any args.

Also, I realized it's probably best that this function is a method and not always unconditionally run. Currently, we use `OwnersRequest` to implement `--changed-*`. Soon, we'll (very likely) use it to implement filesystem specs acting like `--owner-of` for certain goals. These error messages don't make sense in those two contexts.